src/cosalib/ibmcloud.py

```diff
@@ -159,21 +160,109 @@ def ibmcloud_run_ore(build, args):
         f"{args.bucket}/{ibmcloud_object_name}"
     ))
 
-    build.meta[platform] = {
+    build.meta[platform] = [{
         'image': ibmcloud_object_name,
         'bucket': args.bucket,
         'region': region,
         'url': f"https://{url_path}",
-    }
+    }]
     build.meta_write()  # update build metadata
 
 
 def ibmcloud_run_ore_replicate(build, args):
+    build.refresh_meta()
+    platform = args.target
+    if platform == "powervs":
+        ibmcloud_img_data = build.meta.get('powervs', [])
+    else:
+        ibmcloud_img_data = build.meta.get('ibmcloud', [])
+    if len(ibmcloud_img_data) < 1:
+        raise SystemExit(("buildmeta doesn't contain source images. "
+                        "Run buildextend-{platform} first"))
+
+    # define regions - https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-creating-power-virtual-server#creating-service
+    if not args.region:
+        args.region = ['us-east', 'us-south', 'eu-de', 'eu-gb', 'au-syd', 'jp-tok']
+        if platform == "ibmcloud":
+            args.region.extend(['ca-tor', 'ca-mon', 'br-sao', 'jp-osa'])
```

Comment:
am i reading this correctly that the power vs image won't be replicated to ['ca-tor', 'ca-mon', 'br-sao', 'jp-osa']?

Reply:
yes @mkumatag pointed me to: https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-creating-power-virtual-server#creating-service which if i read correctly, the last column indicates where PowerVS instances can be created?

Reply:
ah -- that table is confusing. i believe it's the left-most column. i just poked around in https://cloud.ibm.com/catalog/services/power-systems-virtual-server and double-checked by comparing what's in the drop-down for Location.

Reply:
ok so in the drop down some of them are regions, some datacenters and some zones. so, overall all regions are supported then ?

Reply:
I believe so. Here's what the UPI team has experimented with (scroll down to the table): https://github.com/ocp-power-automation/ocp4-upi-powervs/blob/e4b9898e12c77482f804d63db54a9ac6fac268c7/docs/var.tfvars-doc.md.

Reply:
ok. fixed. although this doesn't matter much as we can override the region list which we will do in the pipeline anyways.